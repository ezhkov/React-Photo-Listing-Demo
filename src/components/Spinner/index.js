import React, { memo } from 'react';
import './Spinner.css';

const Spinner = memo(() => (
  <div className="site-spinner">
    <div className="lds-ring">
      <div>&nbsp;</div>
      <div>&nbsp;</div>
      <div>&nbsp;</div>
      <div>&nbsp;</div>
    </div>
  </div>
));

export default Spinner;
