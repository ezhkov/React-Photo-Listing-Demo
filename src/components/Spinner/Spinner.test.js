import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Spinner from '.';

Enzyme.configure({ adapter: new Adapter() });

describe('Test of component of Spinner', () => {
  it('Test Spinner tags heaving', () => {
    const spinner = shallow(<Spinner />);
    expect(spinner.find('.site-spinner').length).toEqual(1);
    expect(spinner.find('.lds-ring').length).toEqual(1);
    expect(spinner.find('.lds-ring div').length).toEqual(4);
  });

  it('Test Snapshot', () => {
    const spinner = shallow(<Spinner />);
    expect(spinner).toMatchSnapshot();
  });
});
