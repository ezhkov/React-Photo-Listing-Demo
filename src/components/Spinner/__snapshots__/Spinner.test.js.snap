// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of component of Spinner Test Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spinner />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="lds-ring"
      >
        <div>
           
        </div>
        <div>
           
        </div>
        <div>
           
        </div>
        <div>
           
        </div>
      </div>,
      "className": "site-spinner",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
             
          </div>,
          <div>
             
          </div>,
          <div>
             
          </div>,
          <div>
             
          </div>,
        ],
        "className": "lds-ring",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " ",
          },
          "ref": null,
          "rendered": " ",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " ",
          },
          "ref": null,
          "rendered": " ",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " ",
          },
          "ref": null,
          "rendered": " ",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " ",
          },
          "ref": null,
          "rendered": " ",
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="lds-ring"
        >
          <div>
             
          </div>
          <div>
             
          </div>
          <div>
             
          </div>
          <div>
             
          </div>
        </div>,
        "className": "site-spinner",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
               
            </div>,
            <div>
               
            </div>,
            <div>
               
            </div>,
            <div>
               
            </div>,
          ],
          "className": "lds-ring",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " ",
            },
            "ref": null,
            "rendered": " ",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " ",
            },
            "ref": null,
            "rendered": " ",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " ",
            },
            "ref": null,
            "rendered": " ",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " ",
            },
            "ref": null,
            "rendered": " ",
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
